addon/running-transition.js

```diff
@@ -34,6 +34,11 @@ export default class RunningTransition {
 
   _invokeAnimation() {
     return this.animation.run(this.animationContext).then(() => {
+      this.animationContext.oldElement = null;
+      this.animationContext.newElement = null;
+      this.animationContext.older.forEach((entry) => {
+          entry.element = null;
+      });
```

Comment:
I would also expect this not to be necessary, because the RunningTransition should be eligible for GC. If it isn't, that is probably the root bug.